### PR TITLE
Simplify logic a bit

### PR DESCRIPTION
**Pull request by asottile**
- support punctuation
- frozenset of characters
- no longer need `RIGHTHAND`

### CODE REVIEW

**Comment by asottile** [comment]
@theevocater gotta get my golf on!

File: pleasant.py [hunk 1]
```diff
@@ -2,49 +2,33 @@
 from typing import List
 from typing.io import TextIO
 
-LEFTHAND = set(
-    ['q', 'w', 'e', 'r', 't', 'a', 's', 'd', 'f', 'g', 'z', 'x', 'c', 'v', 'b']
-)
-RIGHTHAND = set(['y', 'u', 'i', 'o', 'p', 'h', 'j', 'k', 'l', 'n', 'm'])
+LEFTHAND = frozenset('12345qwertyasdfgzxcvb')
```
Comment (theevocater) [inline pleasant.py]: do you count Y as being on the lefthand? i went with 5 characters from each side. Also do we want punctuation?

File: pleasant.py [hunk 2]
```diff
@@ -2,49 +2,33 @@
...
-            if letter not in hand:
-                found = False
+    for word in dictionary:
+        word = word.lower().strip()
+        left = word[0] in LEFTHAND
+        for c in word[1:]:
+            if (c in LEFTHAND) != left:
                 break
-        if found:
+        else:
```
Comment (theevocater) [inline pleasant.py:31]: i feel like i forgot how to program here

File: pleasant.py (hunk 1 above)
Comment (asottile) [reply pleasant.py]: oh uh, lol.   I cheat at typing -- let me fix that

**Review by theevocater — APPROVED** [review]

**Pull request merged**